pkg/virt-api/webhooks/arm64.go

```diff
@@ -35,40 +35,76 @@ const (
 	defaultCPUModel = v1.CPUModeHostPassthrough
 )
 
-// verifyInvalidSetting verify if VMI spec contain unavailable setting for arm64, check following items:
+// ValidateVirtualMachineInstanceArm64Setting is validation function for validating-webhook
 // 1. if setting bios boot
 // 2. if use uefi secure boot
 // 3. if use host-model for cpu model
-func verifyInvalidSetting(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec) (metav1.StatusCause, bool) {
+// 4. if not use 'sata', 'scsi', 'virtio' as disk bus
+func ValidateVirtualMachineInstanceArm64Setting(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec) []metav1.StatusCause {
+	var statusCauses []metav1.StatusCause
 	if spec.Domain.Firmware != nil && spec.Domain.Firmware.Bootloader != nil {
 		if spec.Domain.Firmware.Bootloader.BIOS != nil {
-			return metav1.StatusCause{
+			statusCauses = append(statusCauses, metav1.StatusCause{
 				Type:    metav1.CauseTypeFieldValueNotSupported,
 				Message: "Arm64 does not support bios boot, please change to uefi boot",
 				Field:   field.Child("domain", "firmware", "bootloader", "bios").String(),
-			}, false
+			})
 		}
 		if spec.Domain.Firmware.Bootloader.EFI != nil {
 			// When EFI is enable, secureboot is enabled by default, so here check two condition
 			// 1 is EFI is enabled without Secureboot setting
 			// 2 is both EFI and Secureboot enabled
 			if spec.Domain.Firmware.Bootloader.EFI.SecureBoot == nil || (spec.Domain.Firmware.Bootloader.EFI.SecureBoot != nil && *spec.Domain.Firmware.Bootloader.EFI.SecureBoot) {
-				return metav1.StatusCause{
+				statusCauses = append(statusCauses, metav1.StatusCause{
 					Type:    metav1.CauseTypeFieldValueNotSupported,
 					Message: "UEFI secure boot is currently not supported on aarch64 Arch",
 					Field:   field.Child("domain", "firmware", "bootloader", "efi", "secureboot").String(),
-				}, false
+				})
 			}
 		}
 	}
 	if spec.Domain.CPU != nil && (&spec.Domain.CPU.Model != nil) && spec.Domain.CPU.Model == "host-model" {
-		return metav1.StatusCause{
+		statusCauses = append(statusCauses, metav1.StatusCause{
 			Type:    metav1.CauseTypeFieldValueNotSupported,
 			Message: "Arm64 not support host model well",
 			Field:   field.Child("domain", "cpu", "model").String(),
-		}, false
+		})
 	}
-	return metav1.StatusCause{}, true
+	if spec.Domain.Devices.Disks != nil {
+		// checkIfBusAvailable: if bus type is nil, sata, virtio, scsi return true, otherwise, return false
+		checkIfBusAvailable := func(bus string) bool {
+			if bus == "" || bus == "sata" || bus == "virtio" || bus == "scsi" {
+				return true
+			}
+			return false
+		}
+
+		for i := range spec.Domain.Devices.Disks {
+			disk := &spec.Domain.Devices.Disks[i].DiskDevice
+			if disk.Disk != nil && !checkIfBusAvailable(disk.Disk.Bus) {
```

Comment:
It looks to me like this function would approve `sata` since `checkIfBusAvailable` seems to return true for `sata` too. That seems to contradict the message which says that `scsi` or `virtio` should be used.